src/renderers/shared/stack/event/ReactSyntheticEventType.js

```diff
@@ -8,14 +8,29 @@
  *
  * Flow type for SyntheticEvent class that includes private properties
  *
- * @providesModule ReactSyntheticEvent
+ * @providesModule ReactSyntheticEventType
  * @flow
  */
 
 'use strict';
 
-export type DispatchConfig = any;
+import type {ReactInstance} from 'ReactInstanceType';
+
+export type DispatchConfig = {
+  dependencies: Array<string>,
+  phasedRegistrationNames?: {
+    bubbled: string,
+    captured: string,
+  },
+  registrationName?: string,
+};
 
 export class ReactSyntheticEvent extends SyntheticEvent {
   dispatchConfig: DispatchConfig;
+  getPooled: (
```

Comment:
Interesting - perhaps PooledClass could export a class that we use as a Flow Mixin  when declaring other classes that later have 'PooledClass.addPooling' called on them. Is that what you meant?

I agree that we should move away from Flow definitions - which we already are with `ReactSyntheticEvent`. If we are thinking of replacing the Flow implementation of React types, we can try and keep a separation between the public API and the internal API of each class, because Flow types should only reflect the public API of React classes.

I can take a stab at both those in a follow-up PR if I'm understanding correctly.